foursight_core/checks/scaling_checks.py

```diff
@@ -0,0 +1,199 @@
+    check = CheckResult(connection, 'datastore_status')
+    db_client = boto3.client('rds', region_name=ecs_utils.COMMON_REGION)
+    es_client = boto3.client('es', region_name=ecs_utils.COMMON_REGION)
+    try:
+        databases = [d['DBInstanceIdentifier'] for d in db_client.describe_db_instances()['DBInstances']]
+        es_clusters = es_client.list_domain_names()['DomainNames']
+    except Exception as e:
+        check.status = 'FAIL'
+        check.summary = f'Could not resolve RDS or ES clusters: {str(e)}'
+        return check
+    else:
+        check.status = 'PASS'
+        check.summary = 'See full output for datastore information'
+        check.full_output = {
+            'rds': databases,
+            'es': es_clusters
+        }
+        return check
+
+
+@check_function()
+def ecs_task_listing(connection, **kwargs):
+    """ Returns information on the available task definitions for launch """
+    check = CheckResult(connection, 'ecs_task_listing')
+    ecs_client = ecs_utils.ECSUtils()
+    try:
+        available_tasks = ecs_client.list_ecs_tasks()
+    except Exception as e:
+        check.status = 'FAIL'
+        check.summary = f'Could not resolve available ECS tasks: {str(e)}'
+        return check
+    else:
+        check.status = 'PASS'
+        check.summary = 'See full output for list of tasks'
+        check.full_output = {
+            'taskDefinitionArns': available_tasks
+        }
+        return check
+
+
+@check_function(cluster_name=None, task_name=None, subnet=None, security_group=None)
+def invoke_ecs_task(connection, **kwargs):
+    """ Invokes an ECS task - intended for use with the deployment action """
+    check = CheckResult(connection, 'invoke_ecs_task')
+    cluster_name = kwargs.get('cluster_name')
+    task_name = kwargs.get('task_name')
+    subnet = kwargs.get('subnet')
+    security_group = kwargs.get('security_group')
+    if not (cluster_name and task_name and subnet and security_group):
+        check.status = 'FAIL'
+        check.summary = 'Did not pass all required parameters: cluster_name, task_name, subnet and security_group'
+        return check
+    else:
+        try:
+            ecs_client = ecs_utils.ECSUtils()
+            # TODO: fix in utils
+            resp = ecs_client.client.run_task(
+                cluster=cluster_name,
+                count=1,
+                taskDefinition=task_name,
+                launchType='FARGATE',  # needs this param
+                networkConfiguration={
+                    'awsvpcConfiguration': {
+                        'subnets': [
+                            subnet
+                        ],
+                        'securityGroups': [
+                            security_group
+                        ]
+                    }
+                })
+        except Exception as e:
+            check.status = 'FAIL'
+            check.summary = f'Got an error trying to run_task: {str(e)}'
+            return check
+        else:
```

Comment:
What is the advantage of using this `else:` over just putting all this stuff at the end of the `try:` body?

This question is not unique this check. I'm just only writing it here. I'm not a big fan of `else:` in `try:` because it lies. It's not really catching things that are not caught in the `except:`. It is catching specifically non-error returns. As such, it's not very perspicuous. Consider:
```
>>> try:
...   raise ValueError("foo")
... except AssertionError:
...   print("assertion problem")
... else:
...   print("something else")
... 
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
ValueError: foo
```
I personally think this makes the intent and behavior clearer. YMMV.
```
>>> try:
...   raise ValueError("foo")
...   print("something else")
... except AssertionError:
...   print("assertion problem")
... 
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
ValueError: foo
```